components/openai/app/openai.app.mjs

```diff
@@ -0,0 +1,160 @@
+import { axios } from "@pipedream/platform";
+
+export default {
+  type: "app",
+  app: "openai",
+  propDefinitions: {
+    completionModelId: {
+      label: "Model",
+      description: "The ID of the model to use for completions. **This action doesn't support the ChatGPT `turbo` models**. Use the **Chat** action for those, instead.",
+      type: "string",
+      async options() {
+        return (await this.getCompletionModels({})).map((model) => model.id);
+      },
+      default: "text-davinci-003",
+    },
+    chatCompletionModelId: {
+      label: "Model",
+      description: "The ID of the model to use for chat completions",
+      type: "string",
+      async options() {
+        return (await this.getChatCompletionModels({})).map((model) => model.id);
+      },
+      default: "gpt-3.5-turbo",
+    },
+    embeddingsModelId: {
+      label: "Model",
+      description: "The ID of the embeddings model to use. OpenAI recommends using `text-embedding-ada-002` for nearly all use cases: \"It's better, cheaper, and simpler to use. [Read the blog post announcement](https://openai.com/blog/new-and-improved-embedding-model)\".",
```

Comment:
Nit, but maybe the trailing " is in the wrong place? Should it read as below? (before linking to the blog post) Not positive though.
```
description: "The ID of the embeddings model to use. OpenAI recommends using `text-embedding-ada-002` for nearly all use cases: \"It's better, cheaper, and simpler to use." [Read the blog post announcement](https://openai.com/blog/new-and-improved-embedding-model)\.",
```

Reply:
that's actually part of the quote from their docs